### PR TITLE
v8.0.0-alpha.0

### PR DESCRIPTION
**Pull request by JCQuintas**

### CODE REVIEW

File: CHANGELOG.md
Comment (JCQuintas) [inline CHANGELOG.md:14]: Is this necessary? A lot of packages have a lot of updates 😅

File: CHANGELOG.md
Comment (JCQuintas) [inline CHANGELOG.md:15]: Doesn't look like it generated the user list...

File: CHANGELOG.md
Comment (JCQuintas) [inline CHANGELOG.md:18]: I thought the `## Changelog` ordered/categorised the changelogs. I was wrong 🙃 

This output is very confusing

File: CHANGELOG.md
Comment (JCQuintas) [inline CHANGELOG.md:43]: @flaviendelangle is all this information necessary? It feels like we are not going straight to the point and that these should be kept in the PR or migration guide.

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:18]: Nop it just adds everything at the top :laughing: 
It would be a nice improvement

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:43]: So far AFAIK we always added the exact same content on the migration guide and on the changelog/release note.

If we want to stop doing it and instead add a summary + a link to the migration guide any time it's a longer section, I'm fine doing it.

WDYT @cherniavskii @alexfauquette @LukasTy @joserodolfofreitas?

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:18]: We should move them in their respective product line inside a `#### Breaking changes` before merging this PR :+1:

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:715]: Should be in the grid section

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:716]:
```suggestion
- [code-infra] Set renovate to automerge `devDependencies` (#13463) @JCQuintas
```

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:709]:
```suggestion
- [core] Add `()` at the name of function name in the doc (#15075) @oliviertassinari
```

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:708]:
```suggestion
- [core] Adjust the `cherry-pick` GitHub actions (#15099) @LukasTy
```

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:573]:
````suggestion
- The `LicenseInfo` object is not exported from the `@mui/x-date-pickers-pro` package anymore.
You can import it from `@mui/x-license` instead:

  ```diff
  -import { LicenseInfo } from '@mui/x-date-pickers-pro';
  +import { LicenseInfo } from '@mui/x-license';
  
  LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
  ```
````

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:14]:
```suggestion
- 🔁 Support [automatic parents and children selection](https://next.mui.com/x/react-tree-view/rich-tree-view/selection/#automatic-parents-and-children-selection) for the Rich Tree View components.
```

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:43]: I guess we could keep the "introduction text" as is and replace all the examples with a link to the migration guide.

Comment (MBilalShafi) [reply CHANGELOG.md:43]: I guess it's only for the `alpha.0` that the BCs take a bit more than expected space, from next versions, since there'll be a release every week, it shouldn't be as long.

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:684]: Done

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:692]:
```suggestion
- [docs] Add migration guide for the removal of `LicenseInfo` from `@mui/x-date-pickers-pro` (#15321) @flaviendelangle 
- [docs] Add section explaining how to keep the selection while filtering (#15185) @arminmeh
```

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:128]: Yes it's generated automatically
But we need to move each BC description in its product line and remove thos "From XXX" :

Comment (flaviendelangle) [reply CHANGELOG.md:128]: In the past, what we did is add a `#### Breaking changes` at the beginning of each product line

For example:

````
### Date and Time Pickers

#### Breaking changes

- The `DesktopDateTimePicker` view rendering has been optimized by using the same technique as for `DesktopDateTimeRangePicker`.
  - The `dateTimeViewRenderers` have been removed in favor of reusing existing time view renderers (`renderTimeViewClock`, `renderDigitalClockTimeView` and `renderMultiSectionDigitalClockTimeView`) and date view renderer (`renderDateViewCalendar`).
  - Passing `renderTimeViewClock` to time view renderers will no longer revert to the old behavior of rendering only date or time view.

#### `@mui/x-date-pickers@7.0.0`

- [fields] Allow to override the separator between the start and the end date in all range fields (#12174) @flaviendelangle
- .... the rest
````

Comment (JCQuintas) [reply CHANGELOG.md:128]: I meant for the automatic PR description

Comment (JCQuintas) [reply CHANGELOG.md:128]: So we could use `## Changelog` here, and `## Migration` elsewhere

Comment (flaviendelangle) [reply CHANGELOG.md:128]: I'm not sure to follow what you mean by "automatic PR description"

Comment (JCQuintas) [reply CHANGELOG.md:128]: Oops. I mean that we add `## Changelog` and it generates the changelog from the PR description. We could have a `## Migration` which would be different from the changelog, to keep it simpler, and could be used to generate a migration doc in the future.

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:627]:
```suggestion
```

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:701]:
`````suggestion
### Charts

#### Breaking changes

- The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object.

  ```jsx
  // Supported by the codemod
  <LineChart legend={{ ... }} />

  // Not supported by the codemod
  const props = { legend: { ... }, ... }
  <Line {...props} />
  ```
  :::
  If you find `slotProps={{ legend: { hidden: true } }}` too long, you can now use the `hideLegend` prop to prevent rendering the legend.
  :::
- The `slots.legend` does not receive the `drawingArea` prop. You can still access your custom legend with the `useDrawingArea()` hook if your custom legend needs it.

`````

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:704]:
```suggestion
- [charts] Introduce `hideLegend` prop (#15277) @alexfauquette
- [charts] Allow `SeriesValueFormatter` to return `null` value (#15057) @clins1994
```

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:566]:
```suggestion
```

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:613]:
```suggestion
```

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:702]:
```suggestion
- The `highlighted` and `faded` properties of highlightScope were deprecated in favor of `highlight` and `fade`.
The deprecated ones are now removed.
- The `xAxisKey`, `yAxisKey`, and `zAxisKey` properties got deprecated in favor of `xAxisId`, `yAxisId`, and `zAxisId`.
- The Pie Chart lost all props and renderer linked to axes because pie chart does not need cartesian axes. If you used it, you can still add them back with composition. Please consider opening an issue to share your use case with us.

#### `@mui/x-charts@8.0.0-alpha.0`
```

File: CHANGELOG.md
Comment (alexfauquette) [reply CHANGELOG.md:128]: We could, the remaining question being "Will people use it?" 😈

But I'm probably too pessimistic on this aspect 😇

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:730]:
```suggestion
- [docs] Add migration guide for the removal of `LicenseInfo` from `@mui/x-date-pickers-pro` (#15321) @flaviendelangle 
- [docs] Add migration guide for the first breaking changes of charts (#15276) @alexfauquette
```

File: CHANGELOG.md
Comment (JCQuintas) [reply CHANGELOG.md]: moved

File: CHANGELOG.md
Comment (JCQuintas) [reply CHANGELOG.md]: i've removed the examples, let me know if the wording is wrong or where we should keep the examples.

 I tried to proofread where I removed them, but you know better than I what was necessary or not

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:70]: This section is empty without the code snippet :laughing: 

For the new DOM structure you can just remove everything under `**Fallback to the non-accessible DOM structure:**`.

And I would add a link to the right section of the migration guide for each breaking change that was shortened.
Something like:

See [Migration guide—New DOM structure for the field](https://next.mui.com/x/migration/migration-pickers-v7/#new-dom-structure-for-the-field) for more details.

File: CHANGELOG.md
Comment (JCQuintas) [reply CHANGELOG.md]: I had to manually remove them 🙃

File: CHANGELOG.md
Comment (JCQuintas) [reply CHANGELOG.md]: Added the migration guide to the changes: https://github.com/mui/mui-x/blob/186313072fd5714ccbadb95042f6a8ed7d5871a0/CHANGELOG.md

Had to manually add them though, and slightly reworked the text in some cases. 

This is not the latest version from master, I'll do another pass when we have a set release date

Comment (flaviendelangle) [reply CHANGELOG.md]: > Had to manually add them though, and slightly reworked the text in some cases.

I'll add them on by PR descriptions for the next BCs :+1:

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:86]:
```suggestion
- [pickers] Remove the re-export from `@mui/x-license` (#14487) @k-rajat19
- [pickers] Unify JSDoc for all the disabled and readOnly props (#15304) @flaviendelangle 
```

File: CHANGELOG.md
Comment (JCQuintas) [reply CHANGELOG.md:52]: My issue with that is that it gets too big for no added value. Add it to each of the migration url and you have a lot of useless info.

It would be fine in the docs or in a blog post though

Comment (JCQuintas) [reply CHANGELOG.md:52]: I used the gh file because it is more static, the `next.mui.com` will make this url useless very soon

Comment (JCQuintas) [reply CHANGELOG.md:52]: > > I used the gh file because it is more static, the next.mui.com will make this url useless very soon
> 
> Not sure to follow you here

Wouldn't `next.mui.com` be "stuck" after we release v8 until we release v9 next year? Or is `next` always up-to-date? 🤔

Comment (flaviendelangle) [reply CHANGELOG.md:52]: > Wouldn't next.mui.com be "stuck" after we release v8 until we release v9 next year? Or is next always up-to-date? 🤔

AFAIK, we will replace all the `next.mui` links with `mui.com` links when releasing major.

Comment (alexfauquette) [reply CHANGELOG.md]: I confirm all links to the docs that appears somewhere else should be prefixed with `next.mui.com` and updated between version.

We had the same issue with JSDocs and agreed to use `https://mui.com` and replace them by `https://next.mui.com` according to the current status of the branch

Comment (flaviendelangle) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md
Comment (JCQuintas) [inline CHANGELOG.md:10]:
```suggestion
<img width="100%" alt="MUI X v8 Alpha is live" src="https://github.com/user-attachments/assets/114cf615-b617-435f-8499-76ac3c26c57b">
```

**Review by flaviendelangle — APPROVED** [review]

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:12]:
```suggestion
_Nov 14, 2024_
```

**Review by LukasTy — APPROVED** [review]
Congrats everyone! 👏 🚀

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:138]: I 'm surprised it's here since the PR target v7.x https://github.com/mui/mui-x/pull/15400/files
```suggestion
```

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:78]: Could you add the improved wording to the migration guide as well? :pray:

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:109]: Oops I missed this one

```suggestion
- The `legend` prop got removed. To pass props to the legend, use `slotProps={{ legend: { ... } }}` instead. This can be automatically done with the codemod as long as the `legend` prop does not come from a destructured object — [Learn more](https://next.mui.com/x/migration/migration-charts-v7/#legend-props-propagation-✅).
```

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:138]: My bad, I manually added all the missing PR and got it wrong on this one

**Comment by DanailH** [comment]
Nice! Let the v8 officially begin! 🚀

**Review by DanailH — APPROVED** [review]

File: CHANGELOG.md
Comment (samuelsycamore) [reply CHANGELOG.md]: Not worth the effort to fix here, but in the future avoid "got removed"—this is a very informal phrasing, and it's unclear who or what did the removing. Instead, use "has been removed"—this implies that we, the maintainers, made the change.

Comment (flaviendelangle) [reply CHANGELOG.md]: cc @alexfauquette

**Pull request merged**
